driver-core/src/main/com/mongodb/internal/async/client/ClientSessionHelper.java

```diff
@@ -61,6 +62,8 @@ void createClientSession(final ClientSessionOptions options, final OperationExec
                 && clusterDescription.getLogicalSessionTimeoutMinutes() != null
                 && clusterDescription.getType() != ClusterType.STANDALONE) {
             callback.onResult(createClientSession(options, executor), null);
+        } else if (mongoClient.getCluster().isClosed()) {
```

Comment:
So this is fixing the second issue that the reporter mentioned, in a subsequent comment?  

This seems racy.  What if the client was closed between evaluating this condition and execution of line 68?  Are we trying to minimize our exposure here, or eliminate it?

Reply:
We are trying to minimize exposure. I don't see a way to fully eliminate it.

Reply:
We can't really remove the race as even the `selectServerAsync` call does an isClosed check and the cluster could be closed after that.

An alternative fix would be to add the callback to the `isTrue` check however its probably best revisiting this pattern when we move to publishers.

Reply:
In that case I would like to remove this check and not claim to have fixed the issue that this is intending to address.

Reply:
I have removed this check.